db/models/code.py

```diff
@@ -0,0 +1,94 @@
+from db import db
+from sqlalchemy.exc import IntegrityError
+
+
+class Code(db.Model):
+    __tablename__ = "Code"
+    id = db.Column(db.String(20), primary_key=True)
+    description = db.Column(db.Text())
+
+    def __init__(self, id=None, description=None):
+        self.id = id
+        self.description = description
+
+    def __repr__(self):
+        return f'Code("{self.id}","{self.description}")'
+
+    def __str__(self):
+        return f"({self.id},{self.description})"
+
+    def as_json(self):
+        return {"id": self.id, "description": self.description}
+
+
+class CodeError(Exception):
+    """Exception raised for errors in Code management
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def __init__(self, message="Sorry, there was a problem, please try later"):
```

Comment:
As this is an internal service, can we be more specific with these errors? They are there to help us operate the service so a more specific error is better in that case.

Reply:
changes made to bubble sql error and overwrite the default error 👍